homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,393 @@
+import voluptuous as vol
+
+from homeassistant.components.switch import ToggleEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN, WiLightDevice
+from .support import wilight_to_hass_trigger, wilight_trigger as wl_trigger
+
+# Bitfield of features supported by the valve switch entities
+SUPPORT_WATERING_TIME = 1
+SUPPORT_PAUSE_TIME = 2
+SUPPORT_TRIGGER_1 = 4
+SUPPORT_TRIGGER_2 = 8
+SUPPORT_TRIGGER_3 = 16
+SUPPORT_TRIGGER_4 = 32
+
+# Attr of features supported by the valve switch entities
+ATTR_WATERING_TIME = "watering_time"
+ATTR_PAUSE_TIME = "pause_time"
+ATTR_TRIGGER_1 = "trigger_1"
+ATTR_TRIGGER_2 = "trigger_2"
+ATTR_TRIGGER_3 = "trigger_3"
+ATTR_TRIGGER_4 = "trigger_4"
+ATTR_TRIGGER_1_DESC = "trigger_1_description"
+ATTR_TRIGGER_2_DESC = "trigger_2_description"
+ATTR_TRIGGER_3_DESC = "trigger_3_description"
+ATTR_TRIGGER_4_DESC = "trigger_4_description"
+
+# Attr of services data supported by the valve switch entities
+ATTR_TRIGGER = "trigger"
+ATTR_TRIGGER_INDEX = "trigger_index"
+
+# Service of features supported by the valve switch entities
+SERVICE_SET_WATERING_TIME = "set_watering_time"
+SERVICE_SET_PAUSE_TIME = "set_pause_time"
+SERVICE_SET_TRIGGER = "set_trigger"
+
+# Range of features supported by the valve switch entities
+RANGE_WATERING_TIME = 1800
+RANGE_PAUSE_TIME = 24
+RANGE_TRIGGER_INDEX = 4
+
+# Service call validation schemas
+VALID_WATERING_TIME = vol.All(
+    vol.Coerce(int), vol.Range(min=1, max=RANGE_WATERING_TIME)
+)
+VALID_PAUSE_TIME = vol.All(vol.Coerce(int), vol.Range(min=1, max=RANGE_PAUSE_TIME))
+VALID_TRIGGER = wl_trigger
+VALID_TRIGGER_INDEX = vol.All(
+    vol.Coerce(int), vol.Range(min=1, max=RANGE_TRIGGER_INDEX)
+)
+
+# Service schemas
+SERVICE_SCHEMA_WATERING_TIME = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_WATERING_TIME): VALID_WATERING_TIME,
+    }
+)
+SERVICE_SCHEMA_PAUSE_TIME = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_PAUSE_TIME): VALID_PAUSE_TIME,
+    }
+)
+SERVICE_SCHEMA_TRIGGER = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_TRIGGER_INDEX): VALID_TRIGGER_INDEX,
+        vol.Required(ATTR_TRIGGER): VALID_TRIGGER,
+    }
+)
+
+# Descriptions of the valve switch entities
+DESC_WATERING = "watering"
+DESC_PAUSE = "pause"
+
+# Icons of the valve switch entities
+ICON_WATERING = "mdi:water"
+ICON_PAUSE = "mdi:pause-circle-outline"
+
+
+def entities_from_discovered_wilight(api_device: Any) -> tuple[Any]:
+    """Parse configuration and add WiLight switch entities."""
+    entities: Any = []
+    for item in api_device.items:
+        if item["type"] == ITEM_SWITCH:
```

Comment:
```suggestion
        if item["type"] != ITEM_SWITCH:
            continue
```

Then outdent below

Reply:
Your suggestion will create a point not covered by the test...

Reply:
I had done it the way you suggested and then I changed it because it was a point not covered by the test.
So the question remains: do I put the continue and start to have a point not covered by the test or not?